docker-plugin/src/main/java/com/nirima/jenkins/plugins/docker/utils/DockerImageName.java

```diff
@@ -0,0 +1,491 @@
+package com.nirima.jenkins.plugins.docker.utils;
+
+import com.google.common.base.Strings;
+import java.util.HashMap;
+import java.util.Map;
+import org.apache.commons.lang.StringUtils;
+
+/**
+ * For constructing and validating docker image names, and their respective parts:
+ * [REGISTRY/][NAMESPACE/]REPOSITORY[:TAG]
+ * 
+ * Based on Docker v1.5 source
+ * @author Joshua Griffiths 08/04/2015
+ */
+
+
+public class DockerImageName {
```

Comment:
You can define multiple patterns with group matches, example https://github.com/jenkinsci/github-plugin/blob/master/src/main/java/com/cloudbees/jenkins/GitHubRepositoryName.java#L26-L48